### PR TITLE
LZ4 as a default compression algorithm

### PR DESCRIPTION
**Pull request by oshadura**

### CODE REVIEW

File: core/zip/src/Bits.h [hunk 1]
```diff
@@ -299,7 +299,7 @@ struct bits_internal_state {
   is done.  LZ4 typically has the worst compression ratios, but much faster decompression
   speeds - sometimes by an order of magnitude.
 */
-enum ECompressionAlgorithm R__ZipMode = 1;
+enum ECompressionAlgorithm R__ZipMode = 4;
```
Comment (amadio) [inline core/zip/src/Bits.h]: Oops, I don't think we can change the global default. We need to find another way.

**Comment by amadio** [comment]
@phsft-bot build

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by oshadura** [comment]
@phsft-bot build!

**Comment by bbockelm** [comment]
[Enable LZ4 as a global for ROOT master, 6.10, 6.08 and 5.34 tags](https://trello.com/c/tEkA4LDN/1-enable-lz4-as-a-global-for-root-master-610-608-and-534-tags)

**Comment by amadio** [comment]
I guess this pull request is obsolete, right? Can someone check and close, please?

**Comment by pcanal** [comment]
Note that we have not yet decided which compression level (of lz4) will be the default.

**Comment by oshadura** [comment]
@pcanal no-no, I know,  I just wanted to test on Jenkins and fix merge conflict...

**Comment by oshadura** [comment]
@phsft-bot build!

**Comment by amadio** [comment]
Please rebase and push again. The problems with master are fixed now.

**Comment by oshadura** [comment]
@phsft-bot build!

File: test/stress.cxx [hunk 2]
```diff
@@ -1213,7 +1216,11 @@ void stress10()
       snprintf(filename,20,"Event_%d.root",file);
       chfile[file] = new TFile(filename,"recreate");
       if (file>=5) {
+#ifdef R__HAS_LZ4
+         chfile[file]->SetCompressionAlgorithm(ROOT::kLZ4);
+#else
          chfile[file]->SetCompressionAlgorithm(ROOT::kLZMA);
+#endif
```
Comment (amadio) [inline test/stress.cxx]: This may be here on purpose to test LZMA, no? Why change to LZ4? Is it just for testing?

Comment (oshadura) [reply test/stress.cxx]: Thanks for good point! it was for testing and I will cleanup!

**Comment by oshadura** [comment]
@phsft-bot build!

**Comment by oshadura** [comment]
Tests are fixed in https://github.com/root-project/roottest/pull/135

File: cmake/modules/RootBuildOptions.cmake [hunk 3]
```diff
@@ -127,6 +127,7 @@ ROOT_BUILD_OPTION(jemalloc OFF "Using the jemalloc allocator")
 ROOT_BUILD_OPTION(krb5 OFF "Kerberos5 support, requires Kerberos libs")
 ROOT_BUILD_OPTION(ldap OFF "LDAP support, requires (Open)LDAP libs")
 ROOT_BUILD_OPTION(libcxx OFF "Build using libc++, requires cxx11 option (MacOS X only, for the time being)")
+ROOT_BUILD_OPTION(lz4 ON "LZ4 compression algorithm as a default, requires liblz4")
```
Comment (pcanal) [inline cmake/modules/RootBuildOptions.cmake]: The option semantic/documentation and its name do not match well.  Usually ItemName=ON means to generate the library for/using this items (see for example the two option above).  Should we use something more lilke: ```lz4_as_default``` or ```compression_default=lz4```?

Comment (oshadura) [reply cmake/modules/RootBuildOptions.cmake]: Thank you for advise, I like both options, maybe second is more flexible, what you think?

Comment (oshadura) [reply cmake/modules/RootBuildOptions.cmake]: I selected first case, but I think  maybe in a case of support more compression algorithms and also changes proposed in https://github.com/root-project/root/pull/1637, I can do a second iteration together with PR#1637

Comment (pcanal) [reply cmake/modules/RootBuildOptions.cmake]: I also think that the 2nd (compression_default=lz4) is better and flexible and would lean for this one.  (If we keep the 1st one we *do need* to change the spelling as building lz4 is not optional)

Comment (oshadura) [reply cmake/modules/RootBuildOptions.cmake]: @pcanal I updated and tested locally according your recommendations..

**Comment by pcanal** [comment]
This Merge Request is still mark as "do not merge".  It looks good to me though (pending double check that the roottest merge-request works with it), is there anything else needed?

**Comment by oshadura** [comment]
@pcanal I have PR with tests https://github.com/root-project/roottest/pull/135, I tested locally with zlib and lz4 (there are still no references for lzma, but I think to add them later, if it will be needed)

**Comment by pcanal** [comment]
roottest changes have been merge.  So relaunching the build for one last check.

**Comment by pcanal** [comment]
@phsft-bot build please.

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
The sole error report seems to be an infrastructure problem of sort.  Merging.

**Pull request merged**